wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -284,6 +285,76 @@ public static function handleAuthorizationEmails(string $op, string $objectName,
     }
   }
 
+  /**
+   * This hook is called after a db write on entities.
+   *
+   * @param string $op
+   *   The type of operation being performed.
+   * @param string $objectName
+   *   The name of the object.
+   * @param int $objectId
+   *   The unique identifier for the object.
+   * @param object $objectRef
+   *   The reference to the object.
+   */
+  public static function linkCollectionCampToContact(string $op, string $objectName, $objectId, &$objectRef) {
+    if ($objectName != 'Eck_Collection_Camp' || !$objectId) {
+      return;
+    }
+
+    $newStatus = $objectRef['Collection_Camp_Core_Details.Status'] ?? '';
+
+    if (!$newStatus) {
+      return;
+    }
+
+    $collectionCamps = EckEntity::get('Collection_Camp', FALSE)
+      ->addSelect('Collection_Camp_Core_Details.Status', 'Collection_Camp_Core_Details.Contact_Id', 'title')
+      ->addWhere('id', '=', $objectId)
+      ->execute();
+
+    $currentCollectionCamp = $collectionCamps->first();
+    $currentStatus = $currentCollectionCamp['Collection_Camp_Core_Details.Status'];
+    $contactId = $currentCollectionCamp['Collection_Camp_Core_Details.Contact_Id'];
+    $collectionCampTitle = $currentCollectionCamp['title'];
+    $collectionCampId = $currentCollectionCamp['id'];
+
+    // Check for status change.
+    if ($currentStatus !== $newStatus) {
+      if ($newStatus === 'authorized') {
+        self::createCollectionCampOrganizeActivity($contactId, $collectionCampTitle, $collectionCampId);
+      }
+    }
+  }
+
+  /**
+   * Log an activity in CiviCRM.
+   */
+  private static function createCollectionCampOrganizeActivity($contactId, $collectionCampTitle, $collectionCampId) {
+    $optionValues = OptionValue::get(TRUE)
+      ->addWhere('option_group_id:name', '=', 'activity_type')
+      ->addWhere('label', '=', 'Organize Collection Camp')
+      ->execute();
+
+    $collectionCampActivityId = $optionValues->first()['value'];
+
+    try {
+      $results = Activity::create(FALSE)
+        ->addValue('subject', $collectionCampTitle)
+        ->addValue('activity_type_id', $collectionCampActivityId)
+        ->addValue('status_id', 2)
+        ->addValue('activity_date_time', date('YmdHis'))
```

Comment:
is this storing the time correctly? shouldn't we use more standard format
```suggestion
        ->addValue('activity_date_time', date('Y-m-d H:i:s'))
```